### PR TITLE
Improved CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ktoso**
Making it Akka specific.

I forgot to explain the tags, will add tomorrow.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,55 +2,45 @@
...
-8. If the code change needs to be applied to other branches as well, create pull requests against those branches which contain the change after rebasing it onto the respective branch and await successful verification by the continuous integration infrastructure; then merge those pull requests.
-
-    Please mark these pull requests with `(for validation)` in the title to make the purpose clear in the pull request list.
+## General Workflow
 
-9. Once everything is said and done, associate the ticket with the “earliest” release milestone (i.e. if back-ported so that it will be in release x.y.z, find the relevant milestone for that release) and close it.
+The below steps are how to get a patch into a main development branch (e.g. `master`). 
+The steps are exactly the same for everyone involved in the project (be it core team, or first time contributor).
+
+1. Make sure there is an issue exists in the [issue tracker](https://github.com/akka/akka/issues) for the work you want to contribute.
```
Comment (patriknw) [inline CONTRIBUTING.md]: s/there is an issue/an issue/

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -198,8 +202,19 @@ the validator to test all projects.
...
 Akka uses [Scalariform](https://github.com/daniel-trinh/scalariform) to enforce some of the code style rules.
 
+### Java style
+
+Java code is currently not automatically reformatted by sbt (expecting to have a plugin to do this soon).
+Thus we ask Java contributions to follow these simple guidelines:
+
+- 2 spaces
+- `{` on same line as method name
+- common sense
```
Comment (patriknw) [inline CONTRIBUTING.md]: isn't there an "official" code convention guide? I know that Sun had one

Comment (ktoso) [inline CONTRIBUTING.md]: Good catch, added:

```
- in all other aspects, follow the [Oracle Java Style Guide](http://www.oracle.com/technetwork/java/codeconvtoc-136057.html)
```

**Comment by patriknw** [comment]
LGTM
Great work, @ktoso !
Now I can finally link to this page without feeling bad.

**Comment by ktoso** [comment]
Added one commit about tags https://github.com/akka/akka/pull/20895/commits/b334db6972a90220d5d54d1a1b42f3d147e58ad1 sanity check and merge please :)

**Pull request merged**